Movie/MovieTests/DetailPresenterTest.swift

```diff
@@ -0,0 +1,120 @@
+    // MARK: - Private Constants
+
+    private enum Constants {
+        static let bazText = "Baz"
+        static let barText = "Bar"
+        static let emptyText = ""
+    }
+
+    // MARK: - Private Property
+
+    private var view: MockDetailView!
+    private var networkService: MockNetworkService!
+    private var presenter: DetailPresenerProtocol!
+    private var router: RouterProtocol!
+    private var id = Constants.emptyText
+    private var detailMovie = DetailMovie()
+    private var actors: [Actor] = []
+    private var data = Data()
+
+    // MARK: - Public Method
+
+    override func setUpWithError() throws {
+        let navController = UINavigationController()
+        let assembly = AssemblyBilder()
+        router = Router(navigationController: navController, assemblyBuilder: assembly)
+    }
+
+    func testSuccesDetailMovie() {
+        let detail = DetailMovie()
+        detail.title = Constants.bazText
+        detail.overview = Constants.barText
+        detailMovie = detail
+        view = MockDetailView()
+        networkService = MockNetworkService(detailMovie: detailMovie)
+
+        presenter = DetailPresenter(
+            view: view,
+            id: Constants.emptyText,
+            networkService: networkService,
+            imageService: ImageService(networkService: networkService),
+            realmService: RealmService(),
+            keychainService: KeychainService(),
+            router: router
+        )
+
+        var catchDetail: DetailMovie?
+        presenter.fetchDetail()
+        networkService.fetchDetail(movieId: Constants.emptyText) { result in
+            switch result {
+            case let .success(detail):
+                catchDetail = detail
+            case let .failure(error):
+                print(error)
+            }
+        }
+
+        XCTAssertNotNil(catchDetail)
+        XCTAssertEqual(catchDetail, detail)
+    }
+
+    func testSuccesActor() {
+        let actor = Actor()
+        actor.name = Constants.bazText
+        actor.character = Constants.barText
+        actors.append(actor)
+        view = MockDetailView()
+        networkService = MockNetworkService(actors: actors)
+        presenter = DetailPresenter(
+            view: view,
+            id: id,
+            networkService: networkService,
+            imageService: ImageService(networkService: networkService),
+            realmService: RealmService(),
+            keychainService: KeychainService(),
+            router: router
+        )
+
+        var catchActor: [Actor]?
+        presenter.fetchActorDetail()
+        networkService.fetchActorDetail(movieId: id) { result in
+            switch result {
+            case let .success(actor):
+                catchActor = actor
+            case let .failure(error):
+                print(error)
+            }
+        }
+
+        XCTAssertNotNil(catchActor?.count)
+        XCTAssertEqual(catchActor, actors)
+    }
+
+    func testSuccesImage() {
```

Comment:
```suggestion
    func testSuccesImage() {
```
Надо еще один тест на проверку failure всегда проверяем все состояния если они есть

Reply:
Для запроса картинки добавил, как понял проверить, что не приходит nil. Написать проверку для других не сложно.